dask-image-quickstart.ipynb

```diff
@@ -0,0 +1,832 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Quickstart to dask-image\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Welcome to the quickstart guide for dask-image."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. [Setting up your environment]('#environment_setup')\n",
+    "1. [Importing dask-image]('#imports')\n",
+    "1. [Getting the example data]('#example_data')\n",
+    "1. [Reading in image data]('#reading_image_data')\n",
+    "    1. [Reading a single image]('#reading_a_single_image')\n",
+    "    1. [Reading multiple iamges]('#reading_multiple_images')\n",
+    "\n",
+    "1. [Applying your own custom function to images]('#applying_custom_function')\n",
+    "    1. [Embarrassingly parallel problems]('#embarrassingly_parallel')\n",
+    "\n",
+    "1. [Joining partial images together]('#joining_images')\n",
+    "\n",
+    "1. [A segmentation analysis pipeline]('#segmentation_pipeline')\n",
+    "    1. [Filtering]('#filtering)\n",
+    "    1. [Segmenting]('#segmenting')\n",
+    "    1. [Analyzing]('#analyzing')\n",
+    "1. [Next steps]('#next_steps')\n",
+    "1. [Cleaning up temporary directories and files]('#cleanup')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='environment_setup'></a>\n",
+    "## Setting up your environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Install Extra Dependencies\n",
+    "\n",
+    "We first install the library scikit-image for easier access to the example image data there."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install scikit-image"
```

Comment:
We're getting the error we get if we update dask to the most recent version.  I suspect that scikit-image is updating dask for us?

Maybe adding `--upgrade-strategy only-if-needed` or `--no-deps` would help here.

Reply:
Are we comfortable just adding scikit-image as a requirement? That would likely improve the experience both for maintainers and users of this example.

Reply:
It looks like the conda-forge [scikit-image package is around 23MB](https://anaconda.org/conda-forge/scikit-image/files)

This is larger than I'm super happy with if there is just one notebook.  I think that for as long as there are only one or two such notebooks I'd prefer to have users pip install in the notebook.  If we invest more heavily here and these notebooks gets a lot of visibility then I think that we would add scikit-image to the docker image.  This is on the border though, if you all feel strongly I'm happy to be overruled. 

This is what we do for other packages like rasterio and tpot.

Reply:
Makes sense. We could vendor the data as noted elsewhere. This may make sense for other reasons too.

It's worth noting that `scikit-image` may already be pulled in as an indirect dependency of `dask-image` currently. Long term we plan to make `scikit-image` an explicit dependency of `dask-image` as there are algorithms we plan to use from it.